gymnasium/envs/external/gym_env.py

```diff
@@ -0,0 +1,212 @@
+    """
+
+    def __init__(
+        self,
+        env_id: Optional[str] = None,
+        make_kwargs: Optional[dict] = None,
+        env: Optional["gym.Env"] = None,
+    ):
+        if GYM_IMPORT_ERROR is not None:
+            raise error.DependencyNotInstalled(
+                f"{GYM_IMPORT_ERROR} (Hint: You need to install gym with `pip install gym` to use gym environments"
+            )
+
+        if make_kwargs is None:
+            make_kwargs = {}
+
+        if env is not None:
+            self.gym_env = env
+        elif env_id is not None:
+            self.gym_env = gym.make(env_id, **make_kwargs)
+        else:
+            raise gymnasium.error.MissingArgument(
+                "Either env_id or env must be provided to create a legacy gym environment."
+            )
+        self.gym_env = _strip_default_wrappers(gym.make(env_id, **make_kwargs))
+
+        self.observation_space = _convert_space(self.gym_env.observation_space)
+        self.action_space = _convert_space(self.gym_env.action_space)
+
+        self.metadata = getattr(self.gym_env, "metadata", {"render_modes": []})
+        self.render_mode = self.gym_env.render_mode
+        self.reward_range = getattr(self.gym_env, "reward_range", None)
+        self.spec = getattr(self.gym_env, "spec", None)
+
+    def reset(
+        self, seed: Optional[int] = None, options: Optional[dict] = None
+    ) -> Tuple[ObsType, dict]:
+        """Resets the environment.
+
+        Args:
+            seed: the seed to reset the environment with
+            options: the options to reset the environment with
+
+        Returns:
+            (observation, info)
+        """
+        super().reset(seed=None)  # We don't need the seed inside gymnasium
+        # Options are ignored
+        return self.gym_env.reset()
+
+    def step(self, action: ActType) -> Tuple[ObsType, float, bool, bool, dict]:
+        """Steps through the environment.
+
+        Args:
+            action: action to step through the environment with
+
+        Returns:
+            (observation, reward, terminated, truncated, info)
+        """
+        return self.gym_env.step(action)
+
+    def render(self):
+        """Renders the environment.
+
+        Returns:
+            The rendering of the environment, depending on the render mode
+        """
+        return self.gym_env.render()
+
+    def close(self):
+        """Closes the environment."""
+        self.gym_env.close()
+
+    def __str__(self):
+        return f"GymEnvironment({self.gym_env})"
+
+    def __repr__(self):
+        return f"GymEnvironment({self.gym_env})"
+
+
+def _strip_default_wrappers(env: "gym.Env") -> "gym.Env":
+    """Strips builtin wrappers from the environment.
+
+    Args:
+        env: the environment to strip builtin wrappers from
+
+    Returns:
+        The environment without builtin wrappers
+    """
+    import gym.wrappers.compatibility  # Cheat because gym doesn't expose it in __init__
```

Comment:
Remove

Reply:
Can't, we'd need to fix it in gym because `compatibility` and the other thing aren't imported in the `__init__.py`

(and we probably want to fix it in gymnasium too)

Reply:
But you are no longer using the compatibility or env_checker wrappers